pkg/aspect/logger/logger.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logger
+
+import (
+	"github.com/golang/protobuf/proto"
+	"istio.io/mixer/pkg/aspect"
+)
+
+type (
+	// Aspect is the interface for adapters that will handle logs data
+	// within the mixer.
+	Aspect interface {
+		aspect.Aspect
+
+		// Log directs a backend adapter to process a batch of
+		// log entries derived from potentially several Report() calls.
+		Log([]Entry) error
+	}
+
+	// Entry is a set of key-value pairs of attributes that together
+	// constitute the data for log entry. The key
+	// is the attribute name, and the value is a typed value for the
+	// named attribute.
+	Entry map[string]interface{}
```

Comment:
the values themselves should already have types, as they come out of the mapper (as I understand it). so, for logs, this might not be that big of a deal, as we will probably have a payload attribute that is ultimately treated as raw text or json and we can count on a json serializer, etc., to build the rest appropriately based on the existing type.

but yes, if the adapters want to do something smart with the inputs, they will need to understand their types.

do inputs (derived attributes) have type descriptors?  where will that information live?

either these smarts need to go into the Manager, so it can extract an entry of typed attributes or the attribute descriptors need to be passed, or we can pass a slimmed down attribute.Bag as the Entry or something else.